File: CSharp/Monitor/Producer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Threading;

namespace ProducerConsumer
{
    /// <summary>
    /// Thread responsável por produzir itens e colocar em um buffer.
    /// </summary>
    public class Producer
    {
        private int id;

        /// <summary>
        /// Buffer na qual será colocado itens
        /// </summary>
	    private Buffer<int> buffer;
	    private bool stop;
        public String Name { get; set; }
	
        /// <summary>
        /// Construtor do Producer
        /// </summary>
        /// <param name="id">Id do produtor</param>
        /// <param name="buffer">buffer na qual será colocado os itens</param>
	    public Producer(int id, Buffer<int> buffer)
        {
		    this.Name = "Producer_" + id;
		    this.buffer = buffer;
		    this.id = id;
		    this.stop = false;
	    }
	
        /// <summary>
        /// Método responsável por produzir e colocar novos itens no buffer.
        /// </summary>
	    public void Run() 
        {
		    int item;
		    Random r = new Random(this.id);
		    while(!stop)
            {
                Thread.Sleep(r.Next(1000));   // Dorme por um tempo randômico

                item = r.Next(1000);    // Produz um número aleatório
                buffer.AddElement(item, this.Name); // Adiciona item produzido no buffer
		    }

		    System.Console.WriteLine(this.Name + " finished!");
	    }	
    }
}

File: Java/PDP/src/producerconsumer/Producer.java
package producerconsumer;

import java.util.Random;

/**
 * Thread responsável por produzir itens e colocar em um buffer.
 * @author Bruno Freitas
 *
 */
public class Producer extends Thread{
	private int id;
	
	/**
	 * Buffer na qual será colocado itens
	 */
	private Buffer<Integer> buffer;
	
	private boolean stop;
	
	/**
	 * Construtor do Producer
	 * @param id - Id do produtor
	 * @param buffer - buffer na qual será colocado os itens
	 */
	public Producer(int id, Buffer<Integer> buffer) {
		super("Producer_" + id);
		this.buffer = buffer;
		this.id = id;
		this.stop = false;
	}
	
	/**
	 * Overrid do método run da classe Thread.
	 * Método responsável por produzir e colocar novos itens no buffer.
	 */
	@Override
	public void run() {
		int item;
		Random r = new Random();
		while(!stop) {
			try {
				Thread.sleep( r.nextInt(1000) );	// Dorme por um tempo randômico
			} catch (Exception e) {
			}
			
			item = r.nextInt(1000);	// Produz um número aleatório
			
			buffer.addElement(item, this.getName());	// Adiciona item produzido no buffer 
			
		}
		System.out.println(this.getName() + " finished!");
	}	
}